src/convertToObject.js

```diff
@@ -6,7 +6,40 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const resObj = {};
+  const cleanedSourceString = sourceString.replace(/\t+/, ' ').trim();
+  const styleStrings = cleanedSourceString.split(';');
+  let currentProperty = '';
+  let currentValue = '';
+
+  styleStrings.forEach((styleString) => {
+    if (styleString.trim() === '') {
+      return;
+    }
+
+    if (styleString.includes(':')) {
+      const index = styleString.indexOf(':');
+      const property = styleString.slice(0, index).trim();
+      const value = styleString.slice(index + 1).trim();
+
+      if (currentProperty) {
+        resObj[currentProperty] = currentValue.trim();
+        currentProperty = '';
+        currentValue = '';
+      }
```

Comment:
[CODE KNOWLEDGE] - The conditional block inside the `forEach` loop seems unnecessary. Since you are splitting the string by semicolons and then processing each style declaration, you should not need to track the `currentProperty` outside of the loop.